scripts/release.py

```diff
@@ -0,0 +1,256 @@
+"""Prepare a Rasa SDK release.
+- creates a release branch
+- creates a new changelog section in CHANGELOG.rst based on all collected changes
+- increases the version number
+- pushes the new branch to GitHub
+"""
+import argparse
+import os
+import sys
+from pathlib import Path
+from subprocess import CalledProcessError, check_call, check_output
+from typing import Text, Set
+
+import questionary
+import semantic_version
+from semantic_version import Version
+
+VERSION_FILE_PATH = "rasa_sdk/version.py"
+
+PYPROJECT_FILE_PATH = "pyproject.toml"
+
+REPO_BASE_URL = "https://github.com/RasaHQ/rasa-sdk"
+
+RELEASE_BRANCH_PREFIX = "prepare-release-"
+
+
+def create_argument_parser() -> argparse.ArgumentParser:
+    """Parse all the command line arguments for the release script."""
+
+    parser = argparse.ArgumentParser(description="prepare the next library release")
+    parser.add_argument(
+        "--next_version",
+        type=str,
+        help="Either next version number or 'major', 'minor', 'patch'",
+    )
+
+    return parser
+
+
+def project_root() -> Path:
+    """Root directory of the project."""
+    return Path(os.path.dirname(__file__)).parent
+
+
+def version_file_path() -> Path:
+    """Path to the python file containing the version number."""
+    return project_root() / VERSION_FILE_PATH
+
+
+def pyproject_file_path() -> Path:
+    """Path to the pyproject.toml."""
+    return project_root() / PYPROJECT_FILE_PATH
+
+
+def write_version_file(version: Text) -> None:
+    """Dump a new version into the python version file."""
+
+    with version_file_path().open("w") as f:
+        f.write(
+            f"# this file will automatically be changed,\n"
+            f"# do not add anything but the version number here!\n"
+            f'__version__ = "{version}"\n'
+        )
+    check_call(["git", "add", str(version_file_path().absolute())])
+
+
+def write_version_to_pyproject(version: Text) -> None:
+    """Dump a new version into the pyproject.toml."""
+
+    import toml
+
+    pyproject_file = pyproject_file_path()
+
+    try:
+        data = toml.load(pyproject_file)
+        data['tool']['poetry']['version'] = version
+        with pyproject_file.open("w") as f:
+            toml.dump(data, f)
+    except (FileNotFoundError, TypeError):
+        print(f"Unable to update {pyproject_file}: file not found.")
```

Comment:
shouldn't we cancel `make release` in case this didn't work?

Reply:
done